### PR TITLE
Feat/error handling

### PR DESCRIPTION
**Pull request by judy0ye**
What I did: Add error handling route and useEffect to rerender page after path changes

Did this break anything?

- [x] No
- [ ]  Yes

Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ]  Bug fix (non-breaking change which fixes an issue)
- [ ]  Refactor(DRY-ing up/ reorganizing code, etc.)
- [ ]  Super small fix (Corrected a typo, removed a comment, etc.)
- [ ]  Skip all the other stuff and briefly explain the fix.

Checklist:

- [ ]  If this code needs to be tested, all tests are passing
- [x]  The code runs locally
- [ ]  There are comments where clarification/ organization is needed
- [x]  The code is DRY. If it's not, I tried my best
- [x]  I reviewed my code before pushing

Notes:

## Contributors to this pull request:
Scotty Brown

### CODE REVIEW

File: src/Components/App/App.js [hunk 1]
```diff
@@ -5,12 +5,18 @@ import Header from '../Header/Header';
...
 import { getAllMovies, getSingleMovie, getMovieVideo } from '../../apiCalls'
 import ErrorHandling from '../ErrorHandling/ErrorHandling'
-import { Route, Routes, useParams } from 'react-router-dom';
+import { Route, Routes, useLocation } from 'react-router-dom';
 
 function App() {
   const [movies, setMovies] = useState({})
   const [singleMovie, setSingleMovie] = useState(null)
   const [error, setError] = useState('')
+  const location = useLocation().pathname
```
Comment (Scotty-Brown) [inline src/Components/App/App.js:14]: Nice job with this side affect to update the error state to null after we navigate away from the error component

File: src/Components/App/App.js [hunk 2]
```diff
@@ -19,6 +25,7 @@ function App() {
   }, [])
 
   function showSingleMovie(id) {
+    // setSingleMovie(null)
```
Comment (Scotty-Brown) [inline src/Components/App/App.js]: Did you mean to comment this out? @judy0ye

File: src/Components/App/App.js [hunk 3]
```diff
@@ -35,6 +42,7 @@ function App() {
       <Routes>
         <Route path='/' element={<Movies movies={movies} showSingleMovie={showSingleMovie}/>} />
         <Route path='/movie/:id' element={<SingleMovie movie={singleMovie} goBack={goBack} />}/> 
+        <Route path='*' element={<ErrorHandling error={error}/>}></Route>
```
Comment (Scotty-Brown) [inline src/Components/App/App.js:45]: Great job with this error handling. Nice way to catch any incorrect urls with the * symbol.

File: src/Components/ErrorHandlingCard/ErrorHandlingCard.js [hunk 4]
```diff
@@ -1,10 +1,13 @@
 import './ErrorHandlingCard.css'
 import grumpyCat from '../../assets/cyrus-chew-Dl39g6QhOIM-unsplash.jpg'
 import PropTypes from 'prop-types'
+import { Link } from 'react-router-dom'
+import { useEffect } from 'react'
 
 function ErrorHandlingCard({error}) {
   return (
     <div className='error'>
+      <Link to='/'>Retry</Link>
```
Comment (Scotty-Brown) [inline src/Components/ErrorHandlingCard/ErrorHandlingCard.js:10]: Nice job implementing this!

**Review by Scotty-Brown — CHANGES_REQUESTED** [review]
Can you check out my comment on line 28 in the app.js file?

File: src/Components/App/App.js (hunk 2 above)
Comment (judy0ye) [reply src/Components/App/App.js]: I meant to keep it in! Good catch!

**Comment by judy0ye** [comment]
Hey @Scotty-Brown, I changed the code snippet you left a comment on- should be good to go!

**Comment by Scotty-Brown** [comment]
> Hey @Scotty-Brown, I changed the code snippet you left a comment on- should be good to go!

Great, Thanks!

**Review by Scotty-Brown — COMMENTED** [review]
👍

**Pull request merged**